runtime/alloc/alloc.cpp

```diff
@@ -51,25 +51,31 @@ void setKoreMemoryFunctionsForGMP() {
    mp_set_memory_functions(koreAllocMP, koreReallocMP, koreFree);
 }
 
-__attribute__ ((always_inline)) void* koreAlloc(size_t requested) {
+#ifdef __clang__
+#define ALWAYS_INLINE __attribute__ ((always_inline))
+#else
+#define ALWAYS_INLINE
```

Comment:
this is a serious problem. If these functions are not inlined, the performance of the entire backend completely bombs to something utterly unacceptable. Gcc supports always inliniing as well, so at the very least you should handle that case.

Reply:
Do we have any benchmark to measure it? When I use that attribute for gcc, it says:
```
error: ‘always_inline’ function might not be inlinable [-Werror=attributes]
```
Maybe disable the warning using pragma GCC ?
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wattributes"
```